src/core/RemotePlugin.cpp

```diff
@@ -62,7 +63,10 @@ void ProcessWatcher::run()
 	{
 		fprintf( stderr,
 				"remote plugin died! invalidating now.\n" );
+
+#ifndef SYNC_WITH_SHM_FIFO
```

Comment:
Could you explain why did you add this check?

Reply:
If you look at the commit that added this change (https://github.com/LMMS/lmms/pull/4421/commits/9db8cbfb31f9b6d3e754887a6280770e3d3e2bcd#diff-b6e06d3d61c47ede770ac49ad9ea8f9e), you can see that I removed an `invalidate()` call guarded by the same check in `processFinished`. I removed it because it prevented receiving messages after the plugin died (related: https://github.com/LMMS/lmms/issues/4103#issuecomment-357974063, #4371, 4fd8ecd7e4084aedbd39900ffe5cfd107ceba1a0). I figured `@jasp00` had a reason to add the check though and simply missed this `invalidate` call, because I couldn't see a point in doing the `invalidate` twice. So I added the checks here because I removed them together with the `invalidate` call in `processFinished`. I'm not sure about this though, it's mostly guesswork. Maybe @DomClark or @Wallacoloo have any insight on this?

Reply:
Since `SYNC_WITH_SHM_FIFO` is defined on Windows, this check will prevent LMMS from invalidating plugin. There's no other calls to `invalidate` on Windows, so it may cause crashes/hangs when remote plugin crashes.
Also, dropping the call in `processFinished` may result in delaying invalidation because it checks process state every 200ms. I guess call was added for immediate invalidation.
I guess we can somehow work around the delay and/or the IPC issue.

Reply:
Should I remove the check? Re-adding the call in `processFinished` will render @Justnope's/@Wallacoloo's 4fd8ecd7e4084aedbd39900ffe5cfd107ceba1a0 useless again.

Reply:
> Should I remove the check?

I think yes. There's no reason to check that here.
> Re-adding the call in `processFinished` (...) 4fd8ecd useless

It does, but can't we work around this? Though it's not necessary for now and might be hard, I don't think it's impossible. I don't want to block merging this because of the call though.

Reply:
> It does, but can't we work around this?

We probably can, but I have no idea how to do it.

Anyway, I'll revert both changes because this PR doesn't depend on them any more since the `VstBadDllFormat` message was replaced by the PE file detection.